linera-core/src/chain_worker/state.rs

```diff
@@ -21,40 +27,444 @@ use linera_chain::{
+                round,
+                published_blobs,
+                callback,
+            } => callback
+                .send(
+                    self.stage_block_execution(block, round, &published_blobs)
+                        .await,
+                )
+                .is_ok(),
+            ChainWorkerRequest::ProcessTimeout {
+                certificate,
+                callback,
+            } => callback
+                .send(self.process_timeout(certificate).await)
+                .is_ok(),
+            ChainWorkerRequest::HandleBlockProposal { proposal, callback } => callback
+                .send(self.handle_block_proposal(proposal).await)
+                .is_ok(),
+            ChainWorkerRequest::ProcessValidatedBlock {
+                certificate,
+                callback,
+            } => callback
+                .send(self.process_validated_block(certificate).await)
+                .is_ok(),
+            ChainWorkerRequest::ProcessConfirmedBlock {
+                certificate,
+                notify_when_messages_are_delivered,
+                callback,
+            } => callback
+                .send(
+                    self.process_confirmed_block(certificate, notify_when_messages_are_delivered)
+                        .await,
+                )
+                .is_ok(),
+            ChainWorkerRequest::ProcessCrossChainUpdate {
+                origin,
+                bundles,
+                callback,
+            } => callback
+                .send(self.process_cross_chain_update(origin, bundles).await)
+                .is_ok(),
+            ChainWorkerRequest::ConfirmUpdatedRecipient {
+                recipient,
+                latest_height,
+                callback,
+            } => callback
+                .send(
+                    self.confirm_updated_recipient(recipient, latest_height)
+                        .await,
+                )
+                .is_ok(),
+            ChainWorkerRequest::HandleChainInfoQuery { query, callback } => callback
+                .send(self.handle_chain_info_query(query).await)
+                .is_ok(),
+            ChainWorkerRequest::DownloadPendingBlob { blob_id, callback } => callback
+                .send(self.download_pending_blob(blob_id).await)
+                .is_ok(),
+            ChainWorkerRequest::HandlePendingBlob { blob, callback } => {
+                callback.send(self.handle_pending_blob(blob).await).is_ok()
+            }
+            ChainWorkerRequest::UpdateReceivedCertificateTrackers {
+                new_trackers,
+                callback,
+            } => callback
+                .send(
+                    self.update_received_certificate_trackers(new_trackers)
+                        .await,
+                )
+                .is_ok(),
+        };
+
+        if !responded {
+            warn!("Callback for `ChainWorkerActor` was dropped before a response was sent");
+        }
+
+        self.chain.rollback();
```

Comment:
This is the new explicit `rollback` call.

Reply:
I'd love a big scary comment here.

Reply:
Done in https://github.com/linera-io/linera-protocol/pull/4388/commits/f41fdf1284534e45845df8e679d85c1f144a4d5c.

Reply:
You wouldn't be able to forget if the API required existed only on the type.

Reply:
But if there's a new request type you'd implement the required API, and that's just as easy (or easier) directly on the chain worker state as it is on the guard.